test/features/resource_availability.feature

```diff
@@ -2,177 +2,250 @@
-        And I press the element with xpath "//form[contains(@class, 'resource-form')]//button[contains(@class, 'btn-primary')]"
-        And I wait for 10 seconds
-        Then I should see "resource created by <User> and is available"
-
-        ###
-        # Create resource that available for non-logged in user with resource_visibility blank.
-        ###
-        And I go to "/dataset/new_resource/contains-de-identified-data-no"
-
-        # Create the resource, with success.
-        Then I execute the script "document.getElementById('field-image-url').value='https://example.com'"
-        And I fill in "name" with "resource created by <User> and is available with blank resource_visibility"
-        And I select "HTML" from "format"
-        And I fill in "description" with "description"
-        And I fill in "size" with "1024"
-        And I press the element with xpath "//form[contains(@class, 'resource-form')]//button[contains(@class, 'btn-primary')]"
-        And I wait for 10 seconds
-        Then I should see "resource created by <User> and is available with blank resource_visibility"
-
-        ###
-        # Create resource that NOT available for non-logged in user.
-        ###
-        And I go to "/dataset/new_resource/contains-de-identified-data-no"
-
-        Then I execute the script "document.getElementById('field-image-url').value='https://example.com'"
-        And I fill in "name" with "resource created by <User> and is NOT available"
-        And I select "HTML" from "format"
-        And I fill in "description" with "description"
-        And I fill in "size" with "1024"
-        And I select "Resource NOT visible/Pending acknowledgement" from "resource_visibility"
-        And I press the element with xpath "//form[contains(@class, 'resource-form')]//button[contains(@class, 'btn-primary')]"
-        And I wait for 10 seconds
-        Then I should see "resource created by <User> and is NOT available"
-
-        # Verify the result as non-logged in user.
-        Given "Unauthenticated" as the persona
-        When I go to "/dataset/contains-de-identified-data-no"
-        Then I should see "resource created by <User> and is available"
-        And I should see "resource created by <User> and is available with blank resource_visibility"
-        And I should not see "resource created by <User> and is NOT available"
-        Then I press the element with xpath "//a[@title='resource created by <User> and is available']"
-        And I should not see "Re-identification risk governance acknowledgement/Resource visibility"
+          Examples: Users
+            | User          |
+            | SysAdmin      |
+            | TestOrgAdmin  |
+            | TestOrgEditor |
 
-        Examples: Users
+    ###
+    # Check the resource  with default values
+    ###
+    Scenario Outline: Check the resource with default values
```

Comment:
These test's are designed to run in sequential order and expect no failing tests on creating datasets. these should be joined with the creation test above or split out into their own file that uses a 'background'

https://pytest-bdd.readthedocs.io/en/latest/#backgrounds

Reply:
Yeh agreed and was unaware of this background feature.
FYI we are not using pytest-bdd but it looks good, we are using behave and they too have a background feature https://behave.readthedocs.io/en/stable/gherkin.html?#background
I will try wrapping up the creation of test data as a background.

Reply:
Unfortunately a background in behave is executed before every scenario of a feature instead of just once per feature.
I refactored the above tests so it is only using one scenario now and created a new step and scenario to create test data.